optuna/storages/journal/_storage.py

```diff
@@ -320,13 +321,29 @@ def set_trial_state_values(
             log["datetime_complete"] = datetime.datetime.now().isoformat(timespec="microseconds")
 
         with self._thread_lock:
+            if state == TrialState.RUNNING:
+                # NOTE(nabenabe): This section is triggered only when we are using `enqueue_trial`
+                # and `GrpcProxyStorage` in distributed optimization setups and solves the issue
+                # https://github.com/optuna/optuna/issues/6084.
+                # When using gRPC, the current thread may already have popped the trial with
+                # trial_id for another process, potentially leading to a false positive in the
+                # return statement of trial_id == _replay_result.owned_trial_id. To eliminate false
+                # positives, we verify whether another process is already evaluating the trial with
+                # trial_id. If True, it means this query does not update the trial state.
+                existing_trial = self._replay_result._trials.get(trial_id)
```

Comment:
Isn't it necessary to call `self._sync_with_backend()`? Since the thin line uses self._reply_result.